### PR TITLE
Modifications in Documentations

### PR DESCRIPTION
**Pull request by arnavrneo**
## Changes done
-  Various typos (including grammatical and some repetitive words) and Markdown correction has been done in **decision-forests** documentations (**Documentation** directory).

- Changes in **keras/core.py**:
     - Few blocks of python code were not marked with markdown. That has been corrected;
     - Some links were not marked as hyperlinks. Those have been corrected;
     - Some typos have been fixed; and
     - **Usage examples** and **Args** of `fit_on_dataset_path` (**Cart Model**) were not formatted properly. They're fixed now.

### CODE REVIEW

File: tensorflow_decision_forests/keras/core.py [hunk 1]
```diff
@@ -301,29 +301,28 @@ class CoreModel(InferenceCoreModel):
     learner: The learning algorithm used to train the model. Possible values
       include (but at not limited to) "LEARNER_*".
     learner_params: Hyper-parameters for the learner. The list of available
-      hyper-parameters is available at
-      https://github.com/google/yggdrasil-decision-forests/blob/main/documentation/learners.md.
+      hyper-parameters is available at [Learners](https://github.com/google/yggdrasil-decision-forests/blob/main/documentation/learners.md).
```
Comment (rstz) [inline tensorflow_decision_forests/keras/core.py:304]: Please revert this, sometimes we prefer a raw link

**Review by rstz — CHANGES_REQUESTED** [review]
LGTM modulo a minor comment.

File: tensorflow_decision_forests/keras/core.py (hunk 1 above)
Comment (arnavrneo) [reply tensorflow_decision_forests/keras/core.py]: Okay. I've reverted the link back to its raw form, can you check it again?

Comment (rstz) [reply tensorflow_decision_forests/keras/core.py]: Thank you, I'll start our internal review process (may take a bit of time, so please be patient)

**Review by rstz — APPROVED** [review]

File: tensorflow_decision_forests/keras/core.py (hunk 1 above)
Comment (arnavrneo) [reply tensorflow_decision_forests/keras/core.py]: Okay. Thanks for your time.

**Pull request merged**